adam/language_specific/english/english_language_generator.py

```diff
@@ -220,6 +219,66 @@ def _translate_object_to_noun(
 
             return dependency_node
 
+        def add_determiner(
+            self,
+            _object: SituationObject,
+            count: int,
+            noun_dependency_node: DependencyTreeToken,
+        ) -> None:
+            possession_relations = [
+                relation
+                for relation in self.situation.relations
+                if relation.relation_type == HAS and relation.second_slot == _object
+            ]
+            if len(possession_relations) > 1:
+                raise RuntimeError("Can not handle multiple possession relations")
+            else:
+                # If speaker possesses the noun
+                if (
+                    len(possession_relations) == 1
+                    and IS_SPEAKER in possession_relations[0].first_slot.properties
+                ):
+                    determiner_node = DependencyTreeToken("my", DETERMINER)
+                    determiner_role = NOMINAL_MODIFIER_POSSESSIVE
+                # If addressee possess the noun
+                elif (
+                    len(possession_relations) == 1
+                    and IS_ADDRESSEE in possession_relations[0].first_slot.properties
+                ):
+                    determiner_node = DependencyTreeToken("your", DETERMINER)
+                    determiner_role = DETERMINER_ROLE
+                # add 's if a non-agent possess the noun
+                elif (
+                    len(possession_relations) == 1
+                    and self.situation.is_dynamic
+                    and possession_relations[0].first_slot
+                    not in only(self.situation.actions).argument_roles_to_fillers[AGENT]
+                ):
+                    token = self.objects_to_dependency_nodes[
```

Comment:
Needs test covered (which, as you note, requires translating the possessing object)